src/muport.js

```diff
@@ -112,21 +112,27 @@ class MuPort {
     if (delegateDids.length !== 3) throw new Error('Must provide exactly 3 DIDs')
     // generate new recoveryNetwork
     const didsPublicKeys = await Promise.all(delegateDids.map(async did => (await MuPort.resolveIdentityDocument(did)).asymEncryptionKey))
-    this.document.recoveryNetwork = await this.keyring.createShares(delegateDids, didsPublicKeys)
+    // seems like the best way to do a deep clone
+    const newDocument = JSON.parse(JSON.stringify(this.document))
+    newDocument.recoveryNetwork = await this.keyring.createShares(delegateDids, didsPublicKeys)
     // save guardians
-    this.document.symEncryptedData.symEncDids = delegateDids.map((did) => this.keyring.symEncrypt(didToBuffer(did)))
-    this.documentHash = await ipfs.addJSONAsync(this.document)
+    newDocument.symEncryptedData = newDocument.symEncryptedData || {}
+    newDocument.symEncryptedData.symEncDids = delegateDids.map((did) => this.keyring.symEncrypt(didToBuffer(did)))
+    let newDocumentHash = await ipfs.addJSONAsync(newDocument)
```

Comment:
Ideally we could compute the hash locally without adding to IPFS here, but that can be an optimization later on.

Reply:
Yeah, Ideally we'd use the real ipfs lib also :)